### PR TITLE
First commit of xCover SDK

### PR DESCRIPTION
**Pull request by uditagarwal**

### CODE REVIEW

File: xcovlib/api/base.py [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+    # All fields defined here will be used for constructing
+    # the URL of the request
+    url_fields = set()
+
+
+    # Initially False, set to True when an instance of the class is created
+    _is_initialized = False
+
+    @classmethod
+    def get(cls, **kwargs):
```
Comment (tyomo4ka) [inline xcovlib/api/base.py]: How about custom attributes which should go in the query? Date for example?

File: xcovlib/http/http_requests.py [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+            _logger.debug("Response was %s:%s", res.status_code, res.content)
+            raise self._exception_for(res.status_code)(
+                res.content.decode('utf-8'), http_code=res.status_code
+            )
+        else:
+            msg = "No response from URL: %s" % res.request.url
+            _logger.error(msg)
+            raise NoResponseError(msg)
+
+    def _send(self, method, path, data):
```
Comment (tyomo4ka) [inline xcovlib/http/http_requests.py]: I don't think we need both _send and _send_json. One should be enough.

File: xcovlib/api/tests/test_quotes.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+class TestQuotes(TestCase):
+
+    def setUp(self):
+        credentials = SignatureAuth(key='XCOVAPIKEY', secret='testsecret')
+        host = '127.0.0.1:8001'
+        conn = HttpRequest(host, auth=credentials)
+        registry.setup({'http_handler': conn})
+
+    def test_quote_creation(self):
```
Comment (tyomo4ka) [inline xcovlib/api/tests/test_quotes.py]: You probably need to mock it with responses. Did you run it with local dev server?

File: xcovlib/http/base.py [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+    }
+
+    success = {
+        200: "OK",
+        201: "Created",
+        204: "Deleted",
+    }
+
+    error_messages = {
+        400: "Bad request: %s",
```
Comment (tyomo4ka) [inline xcovlib/http/base.py]: 422 is missed

File: xcovlib/http/base.py [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+
+class BaseRequest(object):
+    """Base class for http request classes."""
+
+    errors = {
+        400: exceptions.RequestError,
+        401: exceptions.AuthorizationError,
+        403: exceptions.AuthenticationError,
+        404: exceptions.NotFoundError,
+        409: exceptions.ConflictError,
```
Comment (tyomo4ka) [inline xcovlib/http/base.py]: 422 is missed

File: requirements.txt [hunk 6]
```diff
@@ -1,2 +1,11 @@
-nose
-sphinx
\ No newline at end of file
+atomicwrites==1.2.1
```
[the remark this answers is not in the document]
Comment (uditagarwal) [reply requirements.txt]: Doesn't matter, when the library is used and installed from pip, the requirements.txt is not useful, it uses the install_requires from setup.py file. This is just used for development

File: xcovlib/api/base.py (hunk 1 above)
Comment (uditagarwal) [reply xcovlib/api/base.py]: Okay! Working on this

**Pull request merged**